zerver/lib/actions.py

```diff
@@ -1866,16 +1866,14 @@ def notify_reaction_update(user_profile: UserProfile, message: Message,
 
 def do_add_reaction(user_profile: UserProfile, message: Message,
                     emoji_name: str, emoji_code: str, reaction_type: str) -> None:
+    """Should be called while holding a lock on the Message row, to prevent
+    IntegrityErrors.
+    """
```

Comment:
I'd be more explicit here about the lock (since there are lots of various locks) and the things we're preventing with it, e.g.
```
Should be called while holding a FOR UPDATE lock on the Message row, to prevent double-adding the same reaction or race conditions with the message getting deleted concurrently.
```

Reply:
We can mention (e.g. via `access_message(..., lock_message=True)` in the docstring as well.